### PR TITLE
#513 - Make SPARQL query limit configurable by user

### PR DESCRIPTION
**Pull request by MdeBoer95**

### CODE REVIEW

**Comment by MdeBoer95** [comment]
@naveen2507 @reckart 
When I try to use the ``KnowledgeBaseProperties`` like the ``EntityLinkingProperties`` via **@ Resource** in e.g. ``SPARQLQueryStore`` I have the problem that the **@ Resource** can not be used in a static context.
This could be solved by making the ``SPARQLQueryStore`` and ``RDFUtils`` a spring component instead of classes with static methods.
Should I do that or is there a better way to solve this ?

**Comment by reckart** [comment]
> This could be solved by making the SPARQLQueryStore and RDFUtils a spring component instead of classes with static methods.

For 0.6.x, it shouldn't be an issue because the setting is picked up from the `KnowledgeBase` entity anyway.

For 0.5.x, I would suggest to add a limit parameter to the methods in `SPARQLQueryStore` and `RDFUtils` - I expect the code that is calling the methods would not be static and could thus access the `KnowledgeBaseProperties`.

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java [hunk 1]
```diff
@@ -461,22 +465,23 @@ public KBHandle createProperty(KnowledgeBase kb, KBProperty aProperty)
         });
     }
 
-    @Override
-    public Optional<KBProperty> readProperty(KnowledgeBase kb, String aIdentifier)
+    @Override public Optional<KBProperty> readProperty(KnowledgeBase kb, String aIdentifier)
```
Comment (reckart) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java]: `@Override` should be on a line by its own.
`kb` -> `aKB`.

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java [hunk 2]
```diff
@@ -511,7 +516,8 @@ public void deleteProperty(KnowledgeBase kb, KBProperty aType)
         throws QueryEvaluationException
     {
         List<KBHandle> resultList = read(aKB, (conn) -> {
-            String QUERY = SPARQLQueryStore.PROPERTYLIST_QUERY;
+            String QUERY = SPARQLQueryStore
```
Comment (reckart) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java]: `QUERY` is not a constant (`final`). All capital names should only be used for constants and enum values (which are constants).

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/config/KnowledgeBaseProperties.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package de.tudarmstadt.ukp.inception.kb.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationProperties("inception.knowledge-base")
+public class KnowledgeBaseProperties
+{
+    private int sparqlQueryResultLimit = 1000;
```
Comment (reckart) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/config/KnowledgeBaseProperties.java]: Setting should be documented in the asciidoc documentation.

**Pull request merged**